src/app/js/formats/streamgraph/Streamgraph.js

```diff
@@ -0,0 +1,410 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import compose from 'recompose/compose';
+import * as d3 from 'd3';
+
+import { getCssToString, zoomFunction, distinctColors, transformDataIntoMapArray, randomId, getMinMaxValue } from './utils';
+import injectData from '../injectData';
+import exportableToPng from '../exportableToPng';
+
+const styles = {
+    divContainer: {
+        overflow: 'hidden',
+        position: 'relative',
+    },
+    tooltip: {
+        marginTop: 900,
+    },
+    vertical: {
+        marginTop: 60,
+        height: 190,
+        pointerEvents: 'none',
+    },
+    legend: {
+        position: 'relative',
+    },
+    legendItem: {
+        marginTop: 2,
+    },
+    legentItemText: {
+        marginLeft: 5,
+    },
+    legendButton: {
+        height: 15,
+        width: 15,
+    },
+};
+
+class Streamgraph extends PureComponent {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            width: 800,
+            height: 300
+        }
+
+        this.uniqueId = randomId(10);
```

Comment:
By using the React refs API, you can access to the DOM elements of your components without having to setting a uniq ID.